components/tools/OmeroPy/src/omero/tables.py

```diff
@@ -136,6 +136,14 @@ def addOrThrow(self, hdfpath, hdfstorage):
             raise omero.LockTimeout(
                 None, None,
                 "Cannot acquire exclusive lock on: %s" % hdfpath, 0)
+        except IOError, ie:
+            if ie.errno == 9:  # Bad File Descriptor (NFS)
+                hdffile.close()
+                hdffile = hdfstorage.openfile("r")
+                fileno = hdffile.fileno()
```

Comment:
Considering the previous run with less expectations ("r" < "w") passed, I don't expect so, but you're definitely right that something weird could occur. Thanks.